utils.mk

```diff
@@ -163,3 +169,7 @@ standard_rust_check:
 	cargo clippy --all-targets --all-features --release \
 		-- \
 		-D warnings
+
+# Callers should check if this variable is set to avoid including this
+# file multiple times.
+INCLUDED_UTILS_MK = 1
```

Comment:
I assume there's no way to move this check inside `utils.mk`, is that correct? (I'm not aware of any, just making sure we've considered it.)

Reply:
Yes, I considered that. We could do it, nominally we'd end up with `utils.mk` looking like this:

```make
ifeq ($(INCLUDED_UTILS_MK),)

     # << Entire contents of "utils.mk" goes here >>

    INCLUDED_UTILS_MK = 1
endif
```

That would be simpler from the callers perspective (just call `include utils.mk` everywhere you want), but awkward to update `utils.mk` itself.

However, I've just found an alternative which keeps the simpler caller behaviour and avoids the potentially confusing / easy to break `utils.mk` as show above. But there is a catch: we'd need two files, not one:

1. Rename the actual `utils.mk` to something like `private_utils.mk`.
1. Create `utils.mk` as:
    ```make
    ifeq ($(INCLUDED_UTILS_MK),)
        $(eval include private_utils.mk)
    endif
    ```
1. Change all calls to:
    ```make
    include .../utils.mk
    ```

That's a bit more complex, but we could obviously add comments to the `utils.mk` and `private_utils.mk` to explain what's happening.

What do folk think?

Reply:
That's interesting! In my view, given the multiple sites where `utils.mk` is included, I think a solution that doesn't require callers to think of putting the `include` statement within guards would be considerably more robust.  From this perspective, both of the solutions you mention would seem acceptable to me (maybe I'm overlooking something but even wrapping the whole contents of the file in guards doesn't seem too bad to me).